core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -525,6 +527,14 @@ abstract class Chunk[+O] extends Serializable with ChunkPlatform[O] with ChunkRu
 
   override def toString: String =
     iterator.mkString("Chunk(", ", ", ")")
+
+  /** Views this Chunk as a Java unmodifiable List.
+    * Contrary to all methods that start with _"to"_ (e.g. {{toVector}}, {{toArray}}),
+    * this method does not copy data.
+    * As such, this method is mostly intended for `foreach` kind of interop.
+    */
+  def asJava: ju.List[O @uncheckedVariance] =
```

Comment:
What's the motivation for `asJava`? Versus using built-in jdk converters like:

```scala
scala> import scala.jdk.CollectionConverters._

scala> c.iterator.asJava
val res5: java.util.Iterator[Int] = scala.collection.convert.JavaCollectionWrappers$IteratorWrapper@7a2b9987
```

Reply:
@mpilquist copy-pasting from the original issue:
> On top of this, it may also be good to provide an `asJava` facade as well, returning `ju.List`.
> Technically speaking, `asSeq` would allow us to also call the `asJava` the stdlib already provides, but that would be a facade over a facade; not the end of the world but it may be a good tiny optimization.

Reply:
Hm, is this a common need? I don't think I've ever wanted Chunk <=> j.u.List conversion.

Reply:
@mpilquist so, to be completely honest, I am currently needing such conversion which was the main driver for this PR.
However, in the original conversation I had with Arman on **Discord**, before creating the issue, we both concluded that `asSeq` was the most important piece since it would allow `asJava` and more.
Yet, I couldn't shake the idea that having `asJava` directly would be best to avoid the double facade.

Reply:
OK, I don't feel strongly about it so I'm fine with keeping it. Note we dropped some rarely used chunk subtypes in the past -- the various non-byte buffers from java.nio.

Reply:
@mpilquist I am okay with removing the `JavaListAsChunk` and rather just copy-to-array always if you prefer.
However, I do think the `javaList` factory is useful even if just for the sake of being able to `unwrap` the facade.

Reply:
I'm fine either way. Your call.

Reply:
@mpilquist for now I will leave it then, since the code is already there and in the spirit of _"avoid defensive copies"_.
But, will add a comment mentioning that if in the future data show this is rarely used it can be removed.